src/components/DocumentsPage/index.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component } from 'react';
+// components
+import DocumentsEditor from './components/DocumentsEditor';
+import DocumentsManager from './components/DocumentsManager';
+// styles 
+import './styles.css';
+
+class DocumentsPage extends Component {
+  
+  constructor(props) {
+    super();
+    this.state = { documents: {}, document: null, unsaved: {} }
+  };
+
+  next_id = 0;
+  // 'database' documents
+  documents = {};
+
+  onDocumentEdit = (inputName, inputValue) => {
+    this.setState((prevState) => {
+      prevState.document = prevState.document || {}
+      
+      return (prevState.document[inputName] = inputValue);
+    });
+  };
+
+  saveDocument = ({ id, title, content}) => {
+    var document_id = id ? parseInt(id) : this.next_id++;
+
+    this.documents[document_id] = { title, content };
+
+    this.setState((prevState) => {
+      prevState.documents[document_id] = { title, content };
+      delete prevState.unsaved[document_id];
+
+      return { documents: prevState.documents, unsaved: prevState.unsaved, document: { id: null, title: "", content: ""} };
+    });
+    this.forceUpdate();
+  };
+
+  deleteDocument = () => {
+    delete this.documents[this.state.document.id];
+    this.setState((prevState) => {
+      delete prevState.unsaved[prevState.document.id];
+      delete prevState.documents[prevState.document.id];
+
+      return { documents: prevState.documents, unsaved: prevState.unsaved };
+    });
+  };
+
+  addDocumentMode = () => {
+    this.setState((prevState) => {
+      if (prevState.document) {
+        var id = prevState.document ? prevState.document.id : this.next_id++;
+        prevState.unsaved[id] = {...prevState.document, unsaved: true };
+      }
+      prevState.document = { id: null, title: "", content: ""};
+      
+      return { document: prevState.document, unsaved: prevState.unsaved};
+    })
+  };
+
+  editDocumentMode = (id) => {
+    this.setState((prevState) => {
+      var active = prevState.document;
+      
+      if (active.id) {
+        prevState.unsaved[active.id] = { ...active, unsaved: true};
+      }
+      var next = prevState.unsaved[id] || this.documents[id]
+
+      return { document: { ...next, id: id }, unsaved: prevState.unsaved };
+    });
+  };
+
+  searchDocuments = (query) => {
+    var results = {};
+
+    if (!query) {
+      results = this.documents;
+    } else {
+      Object.keys(this.documents).map((dId) =>{
+        var saved = this.documents[dId].title === query || this.documents[dId].content === query;
```

Comment:
use includes or indexOf instead, and make sure to ignore the difference between capital and non-capital letters